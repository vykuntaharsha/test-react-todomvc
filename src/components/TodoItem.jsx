import React, { Component } from 'react';

class TodoItem extends Component {
	constructor(props) {
		super(props);
		this.state = {
			hover: false
		};
		this.handleMouseEnter = this.handleMouseEnter.bind(this);
		this.handleMouseLeave = this.handleMouseLeave.bind(this);
	}

	handleMouseEnter() {
		this.setState({ hover: true });
	}

	handleMouseLeave() {
		this.setState({ hover: false });
	}

	render() {
		const { todo, completeTodo, deleteTodo } = this.props;
		const { hover } = this.state;
		return (
			<li
				className={
					'todo-item ' +
					(todo.completed ? 'completed ' : '') +
					(hover ? 'destroy ' : '')
				}
			>
				<p>{todo.text}</p>
				<input
					className="complete-btn"
					type="checkbox"
					disabled={todo.completed}
					checked={todo.completed}
					onChange={() => completeTodo(todo.id)}
				/>
				<button
					className="delete-btn"
					onClick={() => deleteTodo(todo.id)}
					onMouseEnter={this.handleMouseEnter}
					onMouseLeave={this.handleMouseLeave}
				>
					x
				</button>
			</li>
		);
	}
}

export default TodoItem;
