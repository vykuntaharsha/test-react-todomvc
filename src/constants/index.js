export const todoFilters = {
	ALL: 'show all todos',
	ACTIVE: 'show active todos',
	COMPLETED: 'show completed todos'
};

export const actionTypes = {
	ADD_TODO: 'add todo',
	DELETE_TODO: 'delete todo',
	COMPLETE_TODO: 'complete todo',
	SET_VISIBILITY_FILTER: 'set visibility filter'
};
