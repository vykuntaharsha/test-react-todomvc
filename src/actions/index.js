import { actionTypes } from '../constants';

export const addTodo = text => ({ type: actionTypes.ADD_TODO, text });
export const deleteTodo = id => ({ type: actionTypes.DELETE_TODO, id });
export const completeTodo = id => ({ type: actionTypes.COMPLETE_TODO, id });
export const setVisibilityFilter = filter => ({
	type: actionTypes.SET_VISIBILITY_FILTER,
	filter
});
