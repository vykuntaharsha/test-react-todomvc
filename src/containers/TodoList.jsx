import React from 'react';
import TodoItem from '../components/TodoItem';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { todoFilters } from '../constants';
import { completeTodo, deleteTodo } from '../actions';

const TodoList = ({ todos, actions }) => {
	return (
		<ul className="todo-list">
			<li className="todo-info">
				<p>Todo</p>
				<p>Complete</p>
				<p>Delete</p>
			</li>
			{todos.map(todo => (
				<TodoItem
					key={todo.id}
					todo={todo}
					completeTodo={actions.completeTodo}
					deleteTodo={actions.deleteTodo}
				/>
			))}
		</ul>
	);
};

const mapStateToProps = state => {
	switch (state.visibilityFilter) {
		case todoFilters.ACTIVE:
			return { todos: state.todos.filter(todo => !todo.completed) };
		case todoFilters.COMPLETED:
			return { todos: state.todos.filter(todo => todo.completed) };
		default:
			return { todos: state.todos };
	}
};

const mapDispatchToProps = dispatch => ({
	actions: bindActionCreators({ completeTodo, deleteTodo }, dispatch)
});

export default connect(
	mapStateToProps,
	mapDispatchToProps
)(TodoList);
