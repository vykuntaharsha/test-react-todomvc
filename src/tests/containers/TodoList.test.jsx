import React from 'react';
import TodoList from '../../containers/TodoList';
import configureStore from 'redux-mock-store';
import { shallow } from 'enzyme';
import { todoFilters, actionTypes } from '../../constants';
import { setVisibilityFilter } from '../../actions';
import TodoItem from '../../components/TodoItem';

const setup = filter => {
	const store = configureStore()({
		todos: [
			{
				text: 'pick up groceries',
				completed: false,
				id: 0
			},
			{
				text: 'send an email',
				completed: true,
				id: 1
			}
		],
		visibilityFilter: filter ? filter : todoFilters.ALL
	});
	const wrapper = shallow(<TodoList store={store} />);

	return {
		store,
		wrapper
	};
};

describe('containers', () => {
	describe('TodoList', () => {
		it('should render', () => {
			const { wrapper } = setup();
			expect(wrapper).toMatchSnapshot();
		});

		it('should render active todos when visibilityFilter is ACTIVE', () => {
			const { wrapper } = setup(todoFilters.ACTIVE);
			const todoItems = wrapper.shallow().find(TodoItem);
			todoItems.forEach(item => {
				expect(item.props().todo.completed).toEqual(false);
			});

			expect(todoItems.length).toEqual(1);
		});
	});
});
