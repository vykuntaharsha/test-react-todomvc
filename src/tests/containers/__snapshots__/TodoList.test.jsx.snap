// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers TodoList should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(TodoList)
    store={
        Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "actions": Object {
        "completeTodo": [Function],
        "deleteTodo": [Function],
      },
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeSubscription": Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      },
      "todos": Array [
        Object {
          "completed": false,
          "id": 0,
          "text": "pick up groceries",
        },
        Object {
          "completed": true,
          "id": 1,
          "text": "send an email",
        },
      ],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "actions": Object {
          "completeTodo": [Function],
          "deleteTodo": [Function],
        },
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "storeSubscription": Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "unsubscribe": [Function],
        },
        "todos": Array [
          Object {
            "completed": false,
            "id": 0,
            "text": "pick up groceries",
          },
          Object {
            "completed": true,
            "id": 1,
            "text": "send an email",
          },
        ],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
